library/asn1write.c

```diff
@@ -72,6 +72,19 @@ int mbedtls_asn1_write_tag(unsigned char **p, const unsigned char *start, unsign
     return 1;
 }
 
+static int mbedtls_write_len_and_tag(unsigned char **p,
```

Comment:
Nit: For `static` functions we seem to mostly drop the `mbedtls_` prefix and keep the module name (see `asn1_write_tagged_int()`, `asn1_find_named_data()` etc).

So it would seem better to rename here to `asn1_write_len_and_tag()` for consistency.

Reply:
Actually, I find removing the prefix distracting, and I generally don't like it.

https://mbed-tls.readthedocs.io/en/latest/kb/development/mbedtls-coding-standards/ says "Static functions can use the same naming scheme as non-static functions (`mbedtls_MODULE_frobnicate()` or `psa_frobnicate()`). This is convenient if they are made non-static later, for example for testing. They can also have shorter names if it’s convenient. Keeping the module name (`MODULE_frobnicate()`) as a prefix makes it slightly easier to follow call stacks across modules, but it is not compulsory."

So in this case I would say it's up to the PR author, rather than being better to remove the prefix.

And there are quite a lot of places where the prefix is kept, e.g.

```
library/aes.c:static unsigned mbedtls_aes_rk_offset(uint32_t *buf)
library/aes.c:static int mbedtls_aes_xts_decode_keys(const unsigned char *key,
library/aes.c:static void mbedtls_gf128mul_x_ble(unsigned char r[16],
library/alignment.h:static inline uint16_t mbedtls_bswap16(uint16_t x)
library/alignment.h:static inline uint32_t mbedtls_bswap32(uint32_t x)
library/alignment.h:static inline uint64_t mbedtls_bswap64(uint64_t x)
library/bignum.c:static void mbedtls_mpi_zeroize(mbedtls_mpi_uint *v, size_t n)
library/bignum.c:static int mbedtls_mpi_resize_clear(mbedtls_mpi *X, size_t limbs)
library/bignum.c:static inline mbedtls_mpi_uint mpi_sint_abs(mbedtls_mpi_sint z)
library/bignum.c:static mbedtls_mpi_uint mbedtls_int_div_int(mbedtls_mpi_uint u1,
library/bignum_core.h:static inline size_t mbedtls_mpi_core_montmul_working_limbs(size_t AN_limbs)
library/bignum_mod.c:static int mbedtls_mpi_mod_inv_mont(mbedtls_mpi_mod_residue *X,
library/bignum_mod.c:static int mbedtls_mpi_mod_inv_non_mont(mbedtls_mpi_mod_residue *X,
library/ccm.c:static int mbedtls_ccm_crypt(mbedtls_ccm_context *ctx,
library/ccm.c:static void mbedtls_ccm_clear_state(mbedtls_ccm_context *ctx)
library/ccm.c:static int mbedtls_ccm_compare_tags(const unsigned char *tag1,
library/cipher.c:static int mbedtls_cipher_aead_encrypt(mbedtls_cipher_context_t *ctx,
library/cipher.c:static int mbedtls_cipher_aead_decrypt(mbedtls_cipher_context_t *ctx,
library/common.h:static inline unsigned char *mbedtls_buffer_offset(
library/common.h:static inline const unsigned char *mbedtls_buffer_offset_const(
library/constant_time.c:static inline uint32_t mbedtls_get_unaligned_volatile_uint32(volatile const unsigned char *p)
library/constant_time.c:static size_t mbedtls_ct_size_mask_lt(size_t x,
library/constant_time.c:static unsigned mbedtls_ct_size_gt(size_t x,
library/constant_time.c:static int mbedtls_ct_cond_select_sign(unsigned char condition,
library/constant_time.c:static void mbedtls_ct_mem_move_to_left(void *start,
library/ctr_drbg.c:static int mbedtls_ctr_drbg_reseed_internal(mbedtls_ctr_drbg_context *ctx,
library/debug.c:static void mbedtls_debug_printf_ecdh_internal(const mbedtls_ssl_context *ssl,
library/ecdh.c:static mbedtls_ecp_group_id mbedtls_ecdh_grp_id(
library/ecjpake.c:static int mbedtls_ecjpake_compute_hash(mbedtls_md_type_t md_type,
library/ecjpake.c:static int mbedtls_ecjpake_derive_k(mbedtls_ecjpake_context *ctx,
```

Reply:
Fair enough, but in that case I think we should keep the module name, as is the case with almost all of these examples, so `mbedtls_asn1_write_len_and_tag()`.